src/momento-signer.ts

```diff
@@ -0,0 +1,215 @@
+import {InvalidArgumentError} from '.';
+import {importJWK, JWK, JWTHeaderParameters, KeyLike, SignJWT} from 'jose';
+
+export enum CacheOperation {
+  Get = 'GET',
+  Set = 'SET',
+}
+
+export interface SigningRequest {
+  /**
+   * @param {string} cacheName - Name of the cache the signed token should be allowed to operate on.
+   */
+  cacheName: string;
+  /**
+   * @param {number} expiryEpochSeconds - Timestamp in epoch seconds for
+   * when the signed token should expire.
+   */
+  expiryEpochSeconds: number;
+  /**
+   * @param {string} [cacheKey] - The cache key the signed token is allowed to
+   * access. If not present, auth token is allowed to be used for any key.
+   */
+  cacheKey?: string;
+  /**
+   * @param {CacheOperation} [cacheOperation] - The operation the signed token
+   * should be valid for. If not present the signed token is allowed to be used
+   * for any operation.
+   */
+  cacheOperation?: CacheOperation;
+  /**
+   * @param {number} [ttlSeconds] - The time to live for cache items that this
+   * signed token can set. If not present the signed token is allowed to be
+   * used for set operations with any ttl.
+   */
+  ttlSeconds?: number;
+}
+
+export interface PresignedUrlRequest extends SigningRequest {
+  /**
+   * @param {string} cacheKey - The cache key the presigned
+   * URL should access.
+   */
+  cacheKey: string;
+  /**
+   * @param {CacheOperation} cacheOperation - The operation the presigned
+   * URL should perform on the item in the cache.
+   */
+  cacheOperation: CacheOperation;
+  /**
+   * @param {number} [ttlSeconds] - The time to live for cache items when
+   * performing a SET. Required when operation is CacheOperation.Set.
+   */
+  ttlSeconds?: number;
+}
+
+interface headers extends JWTHeaderParameters {
+  alg: string;
+  typ: string;
+  kid: string;
+}
+
+// `jose.importJWK` returns Promise<KeyLike | Uint8Array>. It doesn't really
+// matter to us which, so this is just an alias for convenience purposes.
+type joseKey = KeyLike | Uint8Array;
+
+/**
+ * Momento Signer
+ *
+ * Features include:
+ * - Create signed jwts for use with HTTP requests to the Momento Simple Cache
+ * Service
+ * - Create presigned URLs to use for HTTP requests to the Momento Simple
+ * Cache Service
+ * @export
+ * @class MomentoSigner
+ */
```

Comment:
Could you include a minimum viable example of the workflow in this header? Would be great in-IDE documentation for users.

Reply:
Sure, I can add that! There's also a PR up in the [examples repo](https://github.com/momentohq/client-sdk-examples/pull/102) with a more extensive example.

Reply:
That's great. As long as we have an example ready to go to demo the usage, you can reference the examples repo one too.